redbox-core/redbox/models/chat.py

```diff
@@ -83,12 +83,17 @@ class ChatResponse(BaseModel):
     route_name: ChatRoute = Field(description="the conversation route taken")
 
 
+class MetadataDetail(BaseModel):
+    input_tokens: dict[str, int] | None = None
+    output_tokens: dict[str, int] | None = None
+
```

Comment:
This is a near copy of: `redbox-core/redbox/models/chain.py:185`. Should one file import the class from other?

Also, are the None options reasonable? (I think I have seen responses with only one of the two token types listed).

Reply:
ah how annoying! unfortunately I think thank laggraph requires us to use TypedDict rather than pydantic so im not sure we can solve this. @wpfl-dbt what do you think?

Reply:
I'm tempted to just leave the two definitions rather than tie ourselves in knots trying to link them - especially as it's only the one repetition.

Reply:
yep, happy with this.